### DISCUSSION TITLE
Error contacting userinfo endpoint

### DISCUSSION THREAD

**Discussion by realSZ27** [9p4/jellyfin-plugin-sso#175] [Q&A] [answered]
I'm using Jellyfin 10.8.13 + Authentik. 
The full error message it gives me is ``Error contacting userinfo endpoint: Forbidden Try logging in again.`` after clicking the login with sso button. It redirects me to and from authentik correctly, and I really have no idea whats causing the problem
<details>
<summary>Jellyfin Logs</summary>
<br>

```bash
jellyfin  | [16:14:54] [INF] [46] Jellyfin.Plugin.SSO_Auth.Api.SSOController: SSO Controller initialized
jellyfin  | [16:14:55] [WRN] [36] Jellyfin.Server.Middleware.ResponseTimeMiddleware: Slow HTTP Response from https://jellyfin.example.com/sso/OID/start/Authentik to my.ip in 0:00:00.6833951 with Status Code 302
jellyfin  | [16:14:56] [INF] [24] Jellyfin.Plugin.SSO_Auth.Api.SSOController: SSO Controller initialized
jellyfin  | [16:15:00] [WRN] [24] Jellyfin.Server.Middleware.ResponseTimeMiddleware: Slow HTTP Response from https://jellyfin.example.com/sso/OID/redirect/Authentik?code=f1a9bdd44cbc4e2c9e7de2f10bd87a39&state=bw1iHbUD5ez8chBgngEOkQ to my.ip in 0:00:04.1407299 with Status Code 400
jellyfin  | [16:15:01] [INF] [36] Jellyfin.Api.Auth.CustomAuthenticationHandler: CustomAuthentication was not authenticated. Failure message: Invalid token.
jellyfin  | [16:15:01] [INF] [36] Jellyfin.Api.Auth.CustomAuthenticationHandler: CustomAuthentication was not authenticated. Failure message: Invalid token.
```

</details>

<details>
<summary>Jellyfin Config</summary>
<br>

![image](https://github.com/9p4/jellyfin-plugin-sso/assets/99378252/0191f4ed-5454-4a27-8eba-c4e475b1cabb)
![image](https://github.com/9p4/jellyfin-plugin-sso/assets/99378252/1938f119-77a3-448d-a3ce-5426ec54b1c2)
![image](https://github.com/9p4/jellyfin-plugin-sso/assets/99378252/b01ec6d1-8b5a-40df-ab56-b538b7c0e7ad)
![image](https://github.com/9p4/jellyfin-plugin-sso/assets/99378252/c3ab3d26-56c8-4c0f-8283-6d54a4812c52)
Everthing not included here is either blank or turned off.
I've also tried turning on ``Do Not Validate OpenID Endpoints (Insecure)`` to no avail.

</details>

<details>
<summary>Authentik Config</summary>
<br>

# Provider
![image](https://github.com/9p4/jellyfin-plugin-sso/assets/99378252/559f3175-45fa-4652-a08d-a8ce18cb14f1)
![image](https://github.com/9p4/jellyfin-plugin-sso/assets/99378252/3bbcbbe1-948b-4f0c-be19-1dbbfef76935)
![image](https://github.com/9p4/jellyfin-plugin-sso/assets/99378252/e5c871fc-728c-4271-a0b4-af5cc163d65f)
# Property Mapping
![image](https://github.com/9p4/jellyfin-plugin-sso/assets/99378252/1c534ea2-56ce-449f-bfd8-6a652040b408)

</details>

Maybe I just missed something, but I haven't been able to figure it out. Also, since I just switched from unstable to stable, this is a fresh install. I also tried it on unstable Jellyfin, and had the same issue.

This Authentik instance works for all the other services I run, so I don't think the problem lies there.

**Comment by 9p4** [comment]
Is Authentik using a GET or a POST to Jellyfin? The plugin only supports an OpenID GET request for the callback.

Furthermore, what is the content of your Authentik's `.well-known/openid-configuration` page? i.e. https://auth.example.com/application/o/jellyfin/.well-known/openid-configuration ?

**Reply by realSZ27** [reply]
I'm not sure how to check if Authentik is using post or get, but here's the output:
```json
{
  "issuer": "https://auth.example.com/application/o/jellyfin/",
  "authorization_endpoint": "https://auth.example.com/application/o/authorize/",
  "token_endpoint": "https://auth.example.com/application/o/token/",
  "userinfo_endpoint": "https://auth.example.com/application/o/userinfo/",
  "end_session_endpoint": "https://auth.example.com/application/o/jellyfin/end-session/",
  "introspection_endpoint": "https://auth.example.com/application/o/introspect/",
  "revocation_endpoint": "https://auth.example.com/application/o/revoke/",
  "device_authorization_endpoint": "https://auth.example.com/application/o/device/",
  "response_types_supported": [
    "code",
    "id_token",
    "id_token token",
    "code token",
    "code id_token",
    "code id_token token"
  ],
  "response_modes_supported": [
    "query",
    "fragment",
    "form_post"
  ],
  "jwks_uri": "https://auth.example.com/application/o/jellyfin/jwks/",
  "grant_types_supported": [
    "authorization_code",
    "refresh_token",
    "implicit",
    "client_credentials",
    "password",
    "urn:ietf:params:oauth:grant-type:device_code"
  ],
  "id_token_signing_alg_values_supported": [
    "RS256"
  ],
  "subject_types_supported": [
    "public"
  ],
  "token_endpoint_auth_methods_supported": [
    "client_secret_post",
    "client_secret_basic"
  ],
  "acr_values_supported": [
    "goauthentik.io/providers/oauth2/default"
  ],
  "scopes_supported": [
    "groups",
    "openid"
  ],
  "request_parameter_supported": false,
  "claims_supported": [
    "sub",
    "iss",
    "aud",
    "exp",
    "iat",
    "auth_time",
    "acr",
    "amr",
    "nonce"
  ],
  "claims_parameter_supported": false,
  "code_challenge_methods_supported": [
    "plain",
    "S256"
  ]
}
```

**Reply by 9p4** [reply]
It all looks good to me, and I can't seem to reproduce your issue. Can your Jellyfin container/machine/VM contact the Authentik endpoint?

**Comment by keesfluitman** [comment, marked as answer]
What I see wrong, is your Authentik setup.
Your Scopes, only have "groups" selected. You need to hold ctrl, and click all scopes you wish to send. i.e. email, openid, profile.

Now, I also have an issue, maybe someone can help me with that ;).
https://github.com/9p4/jellyfin-plugin-sso/discussions/192

**Reply by skieinc** [reply]
Thank you!
